### PR TITLE
Move adding columns into the script. Fixes #40.

### LINKED ISSUE #40: 3.4.1-rc SQL error
`Duplicate column name 'version' SQL=ALTER TABLE `p52jk_weblinks` ADD COLUMN `version` int(10) unsigned NOT NULL DEFAULT '1';`

Update Joomla 3.3.6 to 3.4.1 upload and install the RC package.

I guess we need something like for the drop colums also for the add colums?

### PR DESCRIPTION
**Pull request by wilsonge**
This fixes issue #40: 

`Duplicate column name 'version' SQL=ALTER TABLE `p52jk_weblinks` ADD COLUMN `version` int(10) unsigned NOT NULL DEFAULT '1';`

This message shows when updating Joomla 3.3.6 to 3.4.1 upload and install the weblinks RC package.

### CODE REVIEW

File: src/com_weblinks/script.php [hunk 1]
```diff
@@ -195,4 +198,31 @@ private function dropColumnsIfNeeded()
...
+	 * @since   3.4.1
+	 */
+	private function addColumnsIfNeeded()
+	{
+		$db    = JFactory::getDbo();
+		$table = $db->getTableColumns('#__weblinks');
+
+		if (!array_key_exists('version', $table))
+		{
+			$sql = 'ALTER TABLE ' . $db->qn('#__weblinks') . ' ADD COLUMN ' . $db->qn('version') . ' int(10) unsigned NOT NULL DEFAULT '1';
```
Comment (mbabker) [inline src/com_weblinks/script.php]: Core standardized at one point on the full method names.  Can we please keep that consistency or have we thrown that out the window?

Comment (wilsonge) [inline src/com_weblinks/script.php]: What do you mean by full method names? If it keeps standardisation more than happy to do whatever is needed

Comment (wilsonge) [inline src/com_weblinks/script.php]: I commented and then immediately realised what you meant. Fixed

File: src/com_weblinks/script.php [hunk 2]
```diff
@@ -190,7 +193,34 @@ private function dropColumnsIfNeeded()
...
+	 * @since   3.4.1
+	 */
+	private function addColumnsIfNeeded()
+	{
+		$db    = JFactory::getDbo();
+		$table = $db->getTableColumns('#__weblinks');
+
+		if (!array_key_exists('version', $table))
+		{
+			$sql = 'ALTER TABLE ' . $db->quoteName('#__weblinks') . ' ADD COLUMN ' . $db->quoteName('version') . ' int(10) unsigned NOT NULL DEFAULT '1';
```
Comment (zero-24) [inline src/com_weblinks/script.php]: @wilsonge we get 2 issues here please try:

```
$sql = 'ALTER TABLE ' . $db->quoteName('#__weblinks') . ' ADD COLUMN ' . $db->quoteName('version') . ' int(10) unsigned NOT NULL DEFAULT ' . 1 . '';
``` 

see at the end there we need a closing `'` and points arround the `1`.

Than it works like great :smile:

**Comment by zero-24** [comment]
@test successful with the changes above and in combination with #42 Great work!

**Comment by dongilbert** [comment]
Why is this not using the query builder?

**Comment by mbabker** [comment]
IIRC the database code in general doesn't have support for queries that change table schemas.

**Comment by dongilbert** [comment]
That should be remedied, but probably not in this PR, obv.

**Comment by wilsonge** [comment]
If we could deal with adding and removing columns 99% of extension update SQL files could be killed in one swift chop

**Comment by zero-24** [comment]
Also successful tested by @infograf768 https://github.com/joomla-extensions/weblinks/issues/44#issuecomment-103416428

With my package and https://github.com/joomla-extensions/weblinks/pull/41#discussion_r30539667 included.

Thanks!

File: src/com_weblinks/script.php (hunk 2 above)
Comment (javigomez) [inline src/com_weblinks/script.php]: yup, having the same:

screen shot 2015-05-19 at 13 28 56

screen shot 2015-05-19 at 13 30 26

Comment (wilsonge) [inline src/com_weblinks/script.php]: Should be fixed

**Comment by javigomez** [comment]
Works! 

screen shot 2015-05-19 at 13 46 22

**Comment by javigomez** [comment]
pushed at https://github.com/joomla-extensions/weblinks/commit/c23179f1c74f8a7925b2690d3be7de2da1fb9ff8. Closing this pull.

**Pull request closed without merging**